### ISSUE TITLE
[W&B] Sanitize hyperparameters in WandbLogger

### ISSUE THREAD

**Issue by MrinalJain17** [PyTorchLightning/pytorch-lightning#4731]
## 🐛 Bug

While using Weights & Biases, certain hyperparameters (non-primitives) are not usable in the W&B dashboard as expected. For example, I'm using a list of integers (denoting the filters in each block of a UNet) as a hyperparameter, but this does not show up in the [parallel co-ordinates](https://docs.wandb.com/app/features/panels/parallel-coordinates) section of W&B.

I looked into the base logger and used this function to "sanitize" the hyperparameters before logging to W&B, and it worked correctly. 

https://github.com/PyTorchLightning/pytorch-lightning/blob/ece09f0c5f1af331272a081717cbda4c496dd957/pytorch_lightning/loggers/base.py#L233

### To Reproduce

Unfortunately, the project I'm working on is private for now - so I couldn't share the results. However, if necessary, I could try and use the boring model to reproduce the issue.

### Environment

 - PyTorch Version (e.g., 1.0): 1.7
 - OS (e.g., Linux): Windows
 - How you installed PyTorch (`conda`, `pip`, source): Conda
 - Build command you used (if compiling from source):
 - Python version: 3.7
 - CUDA/cuDNN version: 10.2
 - GPU models and configuration:
 - Any other relevant information:

### Additional context

Following a logic similar to `TensorboardLogger`, I modified the `log_hyperparams()` of `WandbLogger` from:

https://github.com/PyTorchLightning/pytorch-lightning/blob/ece09f0c5f1af331272a081717cbda4c496dd957/pytorch_lightning/loggers/wandb.py#L138-L143

to:

```python
@rank_zero_only
def log_hyperparams(self, params):
    params = self._convert_params(params)
    params = self._flatten_dict(params)
    params = self._sanitize_callable_params(params)
    params = self._sanitize_params(params)
    self.experiment.config.update(params, allow_val_change=True)
```

And the (non-primitive) parameters were logged properly and became usable in the W&B dashboard. I could create a pull request with a fix if this makes sense.

@borisdayma

**Comment by s-rog** [comment]
I'm not very familiar with wandb, what are the accepted types for wandb hparams logging?
(are they consistent with the output types of _sanitize_params: [bool, int, float, str, torch.Tensor]?)

**Comment by borisdayma** [comment]
@MrinalJain17 I'm not really sure on what should be the default behavior for logging arrays.
In some cases, you could for example have a variable size array for example. I'm also scared it would brake currently expected behavior by other users.

It may make sense to log your custom parameters separately using directly `wandb.config['my_param'] = my_value`

**Comment by Borda** [comment]
> I'm not really sure on what should be the default behavior for logging arrays.

how about converting it to a list of lists?

**Comment by awaelchli** [comment]
@MrinalJain17 make sure whatever you are trying to log works with the pure wandb logger without Lightning. If it doesn't, then it means it's not supported by wandb. Our logger interfaces only wrap the third parties and do not add new functionality. 

If I look at the config section of my wandb runs, I see lists correctly displayed, so I'm not convinced we have a bug here. 

The sanitize function converts anything that is not one of (bool, int, float, str, torch.Tensor) to a string, but some loggers support more than these primitive types.

**Comment by MrinalJain17** [comment]
Hi, @awaelchli @Borda @borisdayma Thank you for your responses.

@awaelchli Yes, logging lists do show up in the config section of the W&B dashboard without any issues.
However, they couldn't be used for features like "Parallel Co-ordinates." So, as a workaround, I converted them to lists using lightning's `_sanitize_params()`.

But yeah, I guess this is a W&B issue and not something that lightning should handle implicitly. Closing this issue!
